lib/bosh/workspace/release.rb

```diff
@@ -85,5 +100,38 @@ def release
       end
       release
     end
+
+    def templates_dir
+      File.join(repo.workdir, "templates")
+    end
+
+    def symlink_target(file)
+      if File.readlink(file).start_with?("/")
+        return File.readlink(file)
+      else
+        return File.expand_path(File.join(File.dirname(file), File.readlink(file)))
+      end
+    end
+
+    def submodule_for(file)
+      repo.submodules.each do |submodule|
+        if file.start_with?(File.join(repo.workdir, submodule.path))
+          return submodule
+        end
+      end
+      false
+    end
+
+    def broken_symlink_templates
+      templates = []
+      if FileTest.exists?(templates_dir)
+        Find.find(templates_dir) do |file|
+          if FileTest.symlink?(file) and ! FileTest.exists?(file)
```

Comment:
When only looking for broken symlinks the case of a submodule which needs to be updated is not covered.